minimal/app/ContentView.swift

```diff
@@ -0,0 +1,37 @@
+import Foundation
+import SwiftUI
+import Network
+import ARKit
+
+
+struct ContentView: View {
+    
+    @State var address: String = "127.0.0.1";
+    @State var port: String = "25565";
+    @ObservedObject var delegate: ARDelegate = ARDelegate(client: UDPClient(address: "127.0.0.1", port: "25565")!)
+    
+    var body: some View {
+        ARViewRepresentable(arDelegate: delegate)
+        TextField("ip_address", text: $address)
+            .onSubmit {
+                delegate.client = UDPClient(address: address, port: port)!
+            }
+        TextField("port", text: $port)
+            .onSubmit {
+                delegate.client = UDPClient(address: address, port: port)!
+            }
+    };
+    
+}
+
+struct ARViewRepresentable: UIViewRepresentable {
+    let arDelegate:ARDelegate
+    
+    func makeUIView(context: Context) -> some UIView {
+        let arView = ARSCNView(frame: .zero)
+        arDelegate.setARView(arView)
+        return arView
+    }
+    
+    func updateUIView(_ uiView: UIViewType, context: Context) {}
+}
```

Comment:
Pretty much sums up everything going on, still pretty overwhelming if you're not familiar with SwiftUI or ARKit.

(i.e. UIViewRepresentable?  what is a "delegate"? - I didn't bother figuring it out either, I just make it work)